### PR TITLE
Release prep for 6.0.0

### PR DESCRIPTION
**Pull request by JonRowe**
This will be the final PR merged before releasing 6.0.0, for now its updating the docs and supported versions.

### CODE REVIEW

File: rspec-rails.gemspec [hunk 1]
```diff
@@ -11,7 +11,7 @@ Gem::Specification.new do |s|
   s.homepage    = "https://github.com/rspec/rspec-rails"
   s.summary     = "RSpec for Rails"
   s.description = "rspec-rails is a testing framework for Rails 5+."
-  s.required_ruby_version = ">= 2.2.0"
+  s.required_ruby_version = ">= 2.7.0"
```
Comment (pirj) [inline rspec-rails.gemspec:14]: [Rails 6.1 is 2.5+](https://rubygems.org/gems/rails/versions/6.1.0). Do we want to make it stricter?

**Comment by pirj** [comment]
https://github.com/rspec/rspec-rails/blob/b8f65f3c87454bca393f731cbd724063cef393a8/Gemfile-rails-dependencies#L18 this needs to be flipped to use the 7.0 by default for other RSpec repo sub-builds. And a few small tweaks there.

File: rspec-rails.gemspec (hunk 1 above)
Comment (JonRowe) [reply rspec-rails.gemspec:14]: ALL THE STRICT

Comment (JonRowe) [reply rspec-rails.gemspec:14]: Probably should be 2.5 then, sad times :p

File: README.md [hunk 2]
```diff
@@ -25,21 +26,22 @@ According to [RSpec Rails new versioning strategy][] use:
 [`rspec-rails` 1.x]: https://github.com/dchelimsky/rspec-rails
 [`rspec-rails` 3.x]: https://github.com/rspec/rspec-rails/tree/3-9-maintenance
 [`rspec-rails` 4.x]: https://github.com/rspec/rspec-rails/tree/4-1-maintenance
-[`rspec-rails` 5.x]: https://github.com/rspec/rspec-rails/tree/5-0-maintenance
+[`rspec-rails` 5.x]: https://github.com/rspec/rspec-rails/tree/5-1-maintenance
+[`rspec-rails` 4.x]: https://github.com/rspec/rspec-rails/tree/6-0-maintenance
```
Comment (arkirchner) [inline README.md:30]:
```suggestion
[`rspec-rails` 6.x]: https://github.com/rspec/rspec-rails/tree/6-0-maintenance
```

Comment (JonRowe) [reply README.md]: Ta

**Pull request merged**